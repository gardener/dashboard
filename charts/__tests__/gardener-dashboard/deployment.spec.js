#!/usr/bin/env node
//
// SPDX-FileCopyrightText: 2021 SAP SE or an SAP affiliate company and Gardener contributors
//
// SPDX-License-Identifier: Apache-2.0
//

'use strict'

const { basename } = require('path')
const { helm } = fixtures

const chart = basename(__dirname)
const renderTemplates = (templates, values) => helm.renderChartTemplates(chart, templates, values)

describe('gardener-dashboard', function () {
  describe('deployment', function () {
    let templates

    beforeEach(() => {
      templates = [
        'deployment'
      ]
    })

    it('should render the template with default values', async function () {
      const values = {}
      const documents = await renderTemplates(templates, values)
      expect(documents).toHaveLength(1)
      const [deployment] = documents
      expect(deployment).toMatchSnapshot({
        spec: {
          template: {
            metadata: {
              annotations: {
                'checksum/configmap-gardener-dashboard-config': expect.stringMatching(/[0-9a-f]{64}/),
                'checksum/secret-gardener-dashboard-oidc': expect.stringMatching(/[0-9a-f]{64}/),
                'checksum/secret-gardener-dashboard-sessionSecret': expect.stringMatching(/[0-9a-f]{64}/)
              }
            }
          }
        }
      })
    })
  })
})
