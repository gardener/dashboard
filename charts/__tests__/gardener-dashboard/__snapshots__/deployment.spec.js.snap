// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gardener-dashboard deployment should render the template with default values 1`] = `
Object {
  "apiVersion": "apps/v1",
  "kind": "Deployment",
  "metadata": Object {
    "labels": Object {
      "app": "gardener-dashboard",
      "chart": "gardener-dashboard-0.2.0",
      "heritage": "Helm",
      "release": "dashboard",
    },
    "name": "gardener-dashboard",
    "namespace": "garden",
  },
  "spec": Object {
    "replicas": 1,
    "selector": Object {
      "matchLabels": Object {
        "app": "gardener-dashboard",
        "release": "dashboard",
      },
    },
    "template": Object {
      "metadata": Object {
        "annotations": Object {
          "checksum/configmap-gardener-dashboard-config": StringMatching /\\[0-9a-f\\]\\{64\\}/,
          "checksum/secret-gardener-dashboard-oidc": StringMatching /\\[0-9a-f\\]\\{64\\}/,
          "checksum/secret-gardener-dashboard-sessionSecret": StringMatching /\\[0-9a-f\\]\\{64\\}/,
        },
        "labels": Object {
          "app": "gardener-dashboard",
          "release": "dashboard",
        },
      },
      "spec": Object {
        "containers": Array [
          Object {
            "args": Array [
              "/etc/gardener-dashboard/config.yaml",
            ],
            "env": Array [
              Object {
                "name": "SESSION_SECRET",
                "valueFrom": Object {
                  "secretKeyRef": Object {
                    "key": "keyValue",
                    "name": "gardener-dashboard-sessionsecret",
                  },
                },
              },
              Object {
                "name": "OIDC_CLIENT_ID",
                "valueFrom": Object {
                  "secretKeyRef": Object {
                    "key": "client_id",
                    "name": "gardener-dashboard-oidc",
                  },
                },
              },
              Object {
                "name": "OIDC_CLIENT_SECRET",
                "valueFrom": Object {
                  "secretKeyRef": Object {
                    "key": "client_secret",
                    "name": "gardener-dashboard-oidc",
                  },
                },
              },
            ],
            "image": "eu.gcr.io/gardener-project/gardener/dashboard:1.26.0-dev-4d529c1",
            "imagePullPolicy": "Always",
            "livenessProbe": Object {
              "failureThreshold": 6,
              "initialDelaySeconds": 15,
              "periodSeconds": 20,
              "successThreshold": 1,
              "tcpSocket": Object {
                "port": "http",
              },
              "timeoutSeconds": 5,
            },
            "name": "gardener-dashboard",
            "ports": Array [
              Object {
                "containerPort": 8080,
                "name": "http",
                "protocol": "TCP",
              },
            ],
            "readinessProbe": Object {
              "failureThreshold": 6,
              "httpGet": Object {
                "path": "/healthz",
                "port": "http",
              },
              "initialDelaySeconds": 5,
              "periodSeconds": 10,
              "successThreshold": 1,
              "timeoutSeconds": 5,
            },
            "resources": Object {
              "limits": Object {
                "cpu": "250m",
                "memory": "300Mi",
              },
              "requests": Object {
                "cpu": "100m",
                "memory": "120Mi",
              },
            },
            "volumeMounts": Array [
              Object {
                "mountPath": "/etc/gardener-dashboard",
                "name": "gardener-dashboard-config",
              },
            ],
          },
        ],
        "restartPolicy": "Always",
        "serviceAccountName": "gardener-dashboard",
        "volumes": Array [
          Object {
            "configMap": Object {
              "defaultMode": 420,
              "items": Array [
                Object {
                  "key": "config.yaml",
                  "path": "config.yaml",
                },
              ],
              "name": "gardener-dashboard-configmap",
            },
            "name": "gardener-dashboard-config",
          },
        ],
      },
    },
  },
}
`;
